src/Interactive/EditorFeatures/CSharp/Interactive/CSharpSendToInteractiveSubmissionProvider.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            SyntaxNode expandedNode = GetExecutableSyntaxTreeNode(selectionSpan, source, root, model);
+            return expandedNode != null
+                ? new TextSpan[] { expandedNode.Span }
+                : Array.Empty<TextSpan>();
+        }
+
+        private SyntaxNode GetExecutableSyntaxTreeNode(TextSpan selectionSpan, SourceText source, SyntaxNode root, SemanticModel model)
+        {
+            Tuple<SyntaxToken, SyntaxToken> tokens = GetSelectedTokens(selectionSpan, root);
+            var startToken = tokens.Item1;
+            var endToken = tokens.Item2;
+            if (startToken != endToken && startToken.Span.End > endToken.SpanStart)
+            {
+                return null;
+            }
+
+            // If a selection falls within a single executable statement then execute that statement.
+            var startNode = GetGlobalExecutableStatement(startToken);
+            var endNode = GetGlobalExecutableStatement(endToken);
+            if (startNode == null || endNode == null)
+            {
+                return null;
+            }
+
+            // If one of the nodes is an ancestor of another node return that node.
+            if (startNode.Span.Contains(endNode.Span))
+            {
+                return startNode;
+            }
+            else if (endNode.Span.Contains(startNode.Span))
+            {
+                return endNode;
+            }
+
+            // Selection spans multiple statements.
+            // In this case find common parent and find a span of statements within that parent.
+            var commonNode = root.FindNode(TextSpan.FromBounds(startNode.Span.Start, endNode.Span.End));
+            return commonNode;
+        }
+
+        private static SyntaxNode GetGlobalExecutableStatement(SyntaxToken token)
+        {
+            return GetGlobalExecutableStatement(token.Parent);
+        }
+
+        private static SyntaxNode GetGlobalExecutableStatement(SyntaxNode node)
+        {
+            SyntaxNode candidate = node.GetAncestorOrThis<StatementSyntax>();
+            if (candidate != null)
+            {
+                return candidate;
+            }
+
+            candidate = node.GetAncestorsOrThis<SyntaxNode>()
+                .Where(n => IsGlobalExecutableStatement(n)).FirstOrDefault();
+            if (candidate != null)
+            {
+                return candidate;
+            }
+
+            return null;
+        }
+
+        private static bool IsGlobalExecutableStatement(SyntaxNode node)
+        {
+            var kind = node.Kind();
+            return SyntaxFacts.IsTypeDeclaration(kind)
+                || SyntaxFacts.IsGlobalMemberDeclaration(kind)
+                || node.IsKind(SyntaxKind.UsingDirective);
+        }
+
+        private Tuple<SyntaxToken, SyntaxToken> GetSelectedTokens(TextSpan selectionSpan, SyntaxNode root)
```

Comment:
Unless this is an established pattern in the IDE, consider using out parameters rather than Tuples.